src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http3.cs

```diff
@@ -877,7 +880,8 @@ public async Task Alpn_NonH3_NegotiationFailure()
                 };
 
                 HttpRequestException ex = await Assert.ThrowsAsync<HttpRequestException>(() => client.SendAsync(request).WaitAsync(TimeSpan.FromSeconds(10)));
-                Assert.Contains("ALPN_NEG_FAILURE", ex.Message);
+                AuthenticationException authEx = Assert.IsType<AuthenticationException>(ex.InnerException);
+                Assert.Contains("Application layer protocol negotiation", authEx.Message);
```

Comment:
Just wonder, is it ok to check like that? what if exception message gets localized?

Reply:
I don't know of any other way to check for this specific type of error, so the alternative is to check just for the exception type.

I think I have seen some other test elsewhere that depends on the English localization of the exception messages, but I am not sure where.

Reply:
I checked and there is precious few of such cases in runtime sources (e.g. https://github.com/dotnet/runtime/blob/70669-QuicException/src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/ExceptionTests.cs#L495).

So I will remove the assert